[Enhancement](auditLog) support recording table name and index name in audit log

DarvenDuan opened the pull request:
## Proposed changes

Issue Number: close #xxx

<!--Describe your changes.-->

add a field `queryTables` in edit log to record table name and index name for a query  

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

DarvenDuan committed:
support recording table name and index name in audit log

DarvenDuan commented:
run buildall

yiguolei commented on fe/fe-core/src/main/java/org/apache/doris/plugin/AuditEvent.java:
```diff
@@ -60,6 +60,9 @@ public enum EventType {
     public String catalog = "";
     @AuditField(value = "Db")
     public String db = "";
+    // scan table is formatted as {db}.{table} and  {db}.{table}.{index} if table is OlapTable
```
maybe you also need modify auditlog plugin， because it will load the log into a doris table. I think you add a new field to auditlog
 and the table schema need change.

DarvenDuan replied:
Thanks for your reply, I will implement this latter

DarvenDuan commented:
run p0

DarvenDuan commented:
run p0

DarvenDuan commented:
run p0

yiguolei commented:
You should implement audit logger plugin in same PR, we will release 2.1 now, if this PR is merged and you not implement another PR, then 2.1 could not be released.

yiguolei reviewed (changes requested):
If a user is using doris 2.0 and using audit log plugin to load the log into doris table. If he upgrade doris 2.0 to doris 2.1 and there is a new column tables, what should he do???
I think there is a upgrade process and maybe need a doc.

DarvenDuan committed:
modify audit-pligin.md and fix NPE

DarvenDuan commented:
run buildall

DarvenDuan commented:
> You should implement audit logger plugin in same PR, we will release 2.1 now, if this PR is merged and you not implement another PR, then 2.1 could not be released.

I had add a new field in audit log plugin.

DarvenDuan commented:
> If a user is using doris 2.0 and using audit log plugin to load the log into doris table. If he upgrade doris 2.0 to doris 2.1 and there is a new column tables, what should he do??? I think there is a upgrade process and maybe need a doc.

I modified `audit-plugin.md` and added some notes.

DarvenDuan commented:
run arm

DarvenDuan commented:
run pipelinex_p0

DarvenDuan commented:
run arm

DarvenDuan commented:
run p0

DarvenDuan commented:
run pipelinex

DarvenDuan commented:
run pipelinex_p0 10

DarvenDuan commented:
run compile

DarvenDuan commented:
run p0 10

DarvenDuan commented:
run pipelinex_p0

DarvenDuan commented:
run compile

DarvenDuan commented:
run p0 10

DarvenDuan commented:
run p0

DarvenDuan commented:
run feut

DarvenDuan commented:
run compile

DarvenDuan commented:
run buildall

DarvenDuan commented:
run p0

DarvenDuan commented:
run arm

DarvenDuan commented:
run pipelinex p0

DarvenDuan commented:
run compile

DarvenDuan commented:
run p0

DarvenDuan commented:
run p0 10

DarvenDuan commented:
run p0

DarvenDuan commented:
run p0

DarvenDuan commented:
run p0

DarvenDuan commented:
run feut

DarvenDuan commented:
run beut
